src/service/github.ts

```diff
@@ -1,96 +1,21 @@
-            }
-        }
-        return teams;
-    } catch (error: any) {
-        console.error(`Error: ${error.message}`);
-        return [];
-    }
-};
-
-export const getMembersData = async (memberUrl: string, members: GitHubMembers[] = [], page = 1): Promise<GitHubMembers[]> => {
-    try {
-        const url = `${memberUrl}?page=${page}&per_page=${PER_PAGE}`;
-        const resp = await client.gitHub.getMembers(url) as ApiResponse<GitHubMembers[]>;
-        if (resp.resource) {
-            members = [...members, ...resp.resource];
-            console.log(`${url}, page ${page}, retrieved ${resp.resource.length}`);
-
-            if (resp.resource.length === PER_PAGE) {
-                return await getMembersData(memberUrl, members, page + 1);
-            }
-        }
-        return members;
-    } catch (error: any) {
-        console.error(`Error: ${error.message}`);
-        return [];
-    }
-};
 
-export const getReposData = async (reposUrl: string, repos: GitHubRepos[] = [], page = 1): Promise<GitHubRepos[]> => {
-    try {
-        const url = `${reposUrl}?page=${page}&per_page=${PER_PAGE}`;
-        const resp = await client.gitHub.getRepos(url) as ApiResponse<GitHubRepos[]>;
-        if (resp.resource) {
-            repos = [...repos, ...resp.resource];
-            console.log(`${url}, page ${page}, retrieved ${resp.resource.length}`);
-
-            if (resp.resource.length === PER_PAGE) {
-                return await getReposData(reposUrl, repos, page + 1);
-            }
-        }
-        return repos;
-    } catch (error: any) {
-        console.error(`Error: ${error.message}`);
-        return [];
-    }
-};
-
-export const getMembersPerTeamData = async (membersPerTeamUrl: string, membersPerTeam: GitHubMembersPerTeam[] = [], page = 1): Promise<GitHubMembersPerTeam[]> => {
-    try {
-        const url = `${membersPerTeamUrl}?page=${page}&per_page=${PER_PAGE}`;
-        const resp = await client.gitHub.getMembersPerTeam(url) as ApiResponse<GitHubMembersPerTeam[]>;
-        if (resp.resource) {
-            membersPerTeam = [...membersPerTeam, ...resp.resource];
-            console.log(`${url}, page ${page}, retrieved ${resp.resource.length}`);
-
-            if (resp.resource.length === PER_PAGE) {
-                return await getMembersPerTeamData(membersPerTeamUrl, membersPerTeam, page + 1);
-            }
-        }
-        return membersPerTeam;
-    } catch (error: any) {
-        console.error(`Error: ${error.message}`);
-        return [];
-    }
-};
+import { GITHUB_METHODS_TYPE, GITHUB_TYPES } from "../types/config";
+import { PER_PAGE } from "../config/index";
+import { client } from "./api";
 
-export const getReposPerTeamData = async (reposPerTeamUrl: string, reposPerTeam: GitHubReposPerTeam[] = [], page = 1): Promise<GitHubReposPerTeam[]> => {
+export const getData = async (fnName: GITHUB_METHODS_TYPE, baseUrl: string, data: GITHUB_TYPES[] = [], page = 1): Promise<GITHUB_TYPES[]> => {
```

Comment:
As `getData()` is now a more generalised function, we could use type parameters change the return type per call. This approach would remove the need to use `GITHUB_TYPES` and `as` type casting. Example:

```
// main.ts - provide type parameter here

data[REPOS_KEY] = setReposData(await getData<GitHubRepos>("getRepos", `${GIT_BASE_URL}/${org}/repos`));
```

```
// github.ts - refactor getData() to be generic

// getData() function signature:

export const getData = async <T>(fnName: GITHUB_METHODS_TYPE, baseUrl: string, data: T[] = [], page = 1): Promise<T[]> => {/**/}

// Implementing T parameter:

...
const resp = await client.gitHub[fnName](url) as ApiResponse<T[]>;
return await getData<T>(fnName, baseUrl, data, page + 1);
...
```

Let me know what you think of this idea 👍